### PR TITLE
Eliminated some code in asSeenFrom.

### PR DESCRIPTION
**Pull request by paulp**
Paraphrasing martin, BaseTypeSeqs are now computed lazily
so the cycle "asSeenFrom -> generates BaseTypeSeq -> generates
Refinement of baseType -> needs asSeenFrom" no longer takes
place.  Review by @odersky.

### CODE REVIEW

File: src/reflect/scala/reflect/internal/Types.scala [hunk 1]
```diff
@@ -4325,7 +4313,7 @@ trait Types extends api.Types { self: SymbolTable =>
                 pre1
               }
             } else {
-              toPrefix(base(pre, clazz).prefix, clazz.owner)
+              toPrefix(pre.baseType(clazz).prefix, clazz.owner)
```
Comment (viktorklang) [inline src/reflect/scala/reflect/internal/Types.scala:4316]: I think this needs some more jazz

**Comment by viktorklang** [comment]
Looks badass to me

File: src/reflect/scala/reflect/internal/Types.scala (hunk 1 above)
Comment (paulp) [inline src/reflect/scala/reflect/internal/Types.scala:4316]: A little Thelonious Monk?

Comment (viktorklang) [inline src/reflect/scala/reflect/internal/Types.scala:4316]: J/K, nice catch Paul!

**Comment by odersky** [comment]
This one LGTM. But In fact I suspect the other calls to base can go away as well, and base can be eliminated. @paulp can you try that out?

**Comment by paulp** [comment]
I don't understand the request; this patch already removes the method base.

**Comment by paulp** [comment]
You were perhaps misled by github, which shows a little bit of a patch where someone commented, but that's not the whole patch, just the commented part.  Click "diff" up top to see the whole thing.

**Comment by odersky** [comment]
That's indeed what happened. Happy to see base go!

**Pull request merged**